src/flask_container_scaffold/util.py

```diff
@@ -45,3 +50,30 @@ def _parse_cfg(config):
             settings_dict[key] = config[section][key]
         config_dict.update({section: settings_dict})
     return config_dict
+
+
+def parse_input(logger, obj):
+    """
+    Parses incoming request, returns a serializable object to return
+    to the client in all cases. When there is a failure, the
+    object contains error information.
+    :param Logger logger: Instantiated logger object
+    :param BaseModel obj: An object based ona a pydantic BaseModel to
+                          attempt to parse.
+    :returns: Instantiated object of type obj on success, or QueryResult
+              on failure to parse.
+    """
+    try:
+        if request.is_json:
+            input_args = obj.model_validate_json(json.dumps(request.json))
+        else:
+            if request.args:
+                args = request.args
+            else:
+                args = request.form
+            input_args = obj.model_validate_json(json.dumps(args.to_dict()))
+    except ValidationError as e:
+        logger.error(f"Validation error is: {e}")
+        input_args = BaseApiModel(code=400,
+                                  error=str(e.errors()[0]))
+    return input_args
```

Comment:
question (non-blocking): I wonder if parsed_args might be a better variable name.

Reply:
If that is the general opinion, I am not opposed to changing this. I used this simply because our existing codebases did.